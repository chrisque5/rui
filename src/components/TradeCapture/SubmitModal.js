import PropTypes from 'prop-types';
import { Button, Modal } from 'antd';
import { ExclamationCircleFilled } from '@ant-design/icons';

const SubmitModal = ({ isModalVisible = false, hideModal, onAccept }) => {
  const handleOk = () => {
    hideModal();
    onAccept();
  };

  const handleCancel = () => {
    hideModal();
  };

  return (
    <Modal
      className="confirm-modal"
      visible={isModalVisible}
      onCancel={handleCancel}
      title="Confirm Submit Duplicate Deal"
      footer={[
        <Button data-testid="submit-deal-modal-cancel" key="back" onClick={handleCancel}>Cancel</Button>,
        <Button data-testid="submit-deal-modal-submit" key="submit" type="primary" onClick={handleOk}>Submit Again</Button>,
      ]}
      destroyOnClose
    >
      <div className="modal-body">
        <span className="modal-header">
          <ExclamationCircleFilled />
          You are about to submit a duplicate deal.
        </span>
        <span className="modal-text">
          <p>No changes have been made since the previous deal was submitted.</p>
          <p>If you did not intend to submit a duplicate deal click the &lsquo;Cancel&rsquo; button and change at least one field.</p>
        </span>
      </div>
    </Modal>
  );
};

SubmitModal.propTypes = ({
  isModalVisible: PropTypes.bool.isRequired,
  onAccept: PropTypes.func.isRequired,
  hideModal: PropTypes.func.isRequired,
});

export default SubmitModal;
