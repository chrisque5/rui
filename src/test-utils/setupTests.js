// The below part of the code is to fix the issue
// TypeError: window.matchMedia is not a function
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});
