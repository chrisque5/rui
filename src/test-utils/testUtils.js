/* eslint-disable import/no-extraneous-dependencies, react/prop-types */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Form } from 'antd';

const middlewares = [thunk];
const renderWithForm = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}><Form>{children}</Form></Provider>;

  return render(ui, { wrapper, ...renderOptions });
};

const renderWithProvider = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}>{children}</Provider>;

  return render(ui, { wrapper, ...renderOptions });
};
// re-export everything
export * from '@testing-library/react';
// override render method
export { renderWithForm, renderWithProvider };
