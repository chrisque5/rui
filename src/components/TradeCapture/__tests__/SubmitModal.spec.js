import { fireEvent, render, waitFor } from 'test-utils/testUtils';
import SubmitModal from '../SubmitModal';

const defaultProps = {
  hideModal: jest.fn(),
  onAccept: jest.fn(),
};

afterEach(() => {
  jest.clearAllMocks();
});

test('<SubmitModal /> renders snapshot as expected', () => {
  const { container } = render(<SubmitModal {...defaultProps} />);
  expect(container).toMatchSnapshot();
});

test('<SubmitModal /> renders snapshot as expected when isModalVisible is true', () => {
  const props = {
    ...defaultProps,
    isModalVisible: true,
  };

  const { container } = render(<SubmitModal {...props} />);
  expect(container).toMatchSnapshot();
});

test('<SubmitModal /> calls hideModal() on cancel', async () => {
  const props = {
    ...defaultProps,
    isModalVisible: true,
  };

  const { getByTestId } = render(<SubmitModal {...props} />);
  const cancelButton = await waitFor(() => getByTestId('submit-deal-modal-cancel'));

  fireEvent.click(cancelButton);

  expect(defaultProps.hideModal).toHaveBeenCalledTimes(1);
  expect(defaultProps.onAccept).not.toHaveBeenCalled();
});

test('<SubmitModal /> calls hideModal() and onAccept() on submit', async () => {
  const props = {
    ...defaultProps,
    isModalVisible: true,
  };

  const { getByTestId } = render(<SubmitModal {...props} />);
  const submitButton = await waitFor(() => getByTestId('submit-deal-modal-submit'));

  fireEvent.click(submitButton);

  expect(defaultProps.hideModal).toHaveBeenCalledTimes(1);
  expect(defaultProps.onAccept).toHaveBeenCalledTimes(1);
});
