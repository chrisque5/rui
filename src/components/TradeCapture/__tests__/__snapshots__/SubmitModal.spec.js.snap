// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitModal /> renders snapshot as expected 1`] = `<div />`;

exports[`<SubmitModal /> renders snapshot as expected when isModalVisible is true 1`] = `<div />`;
